distributed/protocol/serialize.py

```diff
@@ -135,6 +135,16 @@ def serialize(x, serializers=None, on_error="message", context=None):
     if isinstance(x, Serialized):
         return x.header, x.frames
 
+    if isinstance(x, tuple):
+        t = [
+            serialize(i, serializers=serializers, on_error=on_error, context=context)
+            for i in x
+        ]
+        headers = [i[0] for i in t]
+        headers = {"sub-headers": headers, "is-collection": True}
+        frames = [i[1] for i in t]
+        return headers, frames
```

Comment:
*If* we do this then I would rather do it on a serializer-by-serialier basis by adding a new implementation for `dask_seriailize/dask_deserialize` and `cuda_serialize/cuda_deserialize`.

```python
@cuda_serialize.register(tuple)
def cuda_serialize_tuple(obj):
    ...

@cuda_deserialize.register(tuple)
def cuda_deserialize_tuple(header, frames):
    ...
```

However having this on by default in `dask_serialize` may also cause performance problems with workloads that pass around a lot of tuples/lists of small Python objects.  This is particularly common for text processing and JSON-like data processing with systems like Dask Bag.

cc @jcrist and @TomAugspurger , in case they want to weigh in on if we should unpack core Python collections in `dask_serialize`